core/reactor_common.c

```diff
@@ -1726,48 +1838,72 @@ int process_args(int argc, char* argv[]) {
                 usage(argc, argv);
                 return 0;
             }
-        } else if (strcmp(argv[i], "-k") == 0 || strcmp(argv[i], "--keepalive") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-k") == 0 || strcmp(arg, "--keepalive") == 0) {
+            if (argc < i + 1) {
                 error_print("--keepalive needs a boolean.");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
-            char* keep_spec = argv[i];
+            char* keep_spec = argv[i++];
             if (strcmp(keep_spec, "true") == 0) {
                 keepalive_specified = true;
             } else if (strcmp(keep_spec, "false") == 0) {
                 keepalive_specified = false;
             } else {
                 error_print("Invalid value for --keepalive: %s", keep_spec);
             }
-        } else if (strcmp(argv[i], "-t") == 0 || strcmp(argv[i], "--threads") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-t") == 0 || strcmp(arg, "--threads") == 0) {
+            if (argc < i + 1) {
                 error_print("--threads needs an integer argument.s");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
             char* threads_spec = argv[i++];
             int num_threads = atoi(threads_spec);
             if (num_threads <= 0) {
                 error_print("Invalid value for --threads: %s. Using 1.", threads_spec);
                 num_threads = 1;
             }
             _lf_number_of_threads = (unsigned int)num_threads;
-        } else if (strcmp(argv[i], "-i") == 0 || strcmp(argv[i], "--id") == 0) {
-            if (argc < i + 2) {
+        } else if (strcmp(arg, "-i") == 0 || strcmp(arg, "--id") == 0) {
+            if (argc < i + 1) {
                 error_print("--id needs a string argument.");
                 usage(argc, argv);
                 return 0;
             }
-            i++;
-            info_print("Federation ID for executable %s: %s", argv[0], argv[i]);
-            federation_id = argv[i++];
-        } else if (strcmp(argv[i], "--ros-args") == 0) {
+            federation_metadata.federation_id = argv[i++];
+            info_print("Federation ID for executable %s: %s", argv[0], federation_metadata.federation_id);
+        } else if (strcmp(arg, "--rti") == 0) {
+            if (argc < i + 1) {
+                error_print("--rti needs a string argument.");
+                usage(argc, argv);
+                return 0;
+            }
+            parse_rti_code_t code = parse_rti_addr(argv[i++]);
+            if (code != SUCCESS) {
+                switch (code) {
+                    case INVALID_HOST:
```

Comment:
It seems like `INVALID_HOST` and `INVALID_USER` are never returned. Perhaps `extract_rti_addr_info` is the best place to detect these two errors.